### ISSUE TITLE
Do we need full-res geotiff to be produced?

### ISSUE THREAD

**Issue by mcgarth** [GeoscienceAustralia/PyRate#180]
This was never an issue until I tried running prepifg over full-size Sentinel-1 interferograms.

When running prepifg, firstly the Gamma format interferogram is read and written out to a geotiff. Secondly this new geotiff is read in and cropped/multi-looked.

I think full-res geotiff output should be optional. It would save run-time and disk space if the full-res geotiff generation was skipped, particularly for Sentinel-1 datasets (5.5G per ifg)

**Comment by basaks** [comment]
Another option you could explore is to produce the tifs in a compressed form. While the compression ratio depends on the data, often the compressed file size can be 2x-10x smaller. [Here is one example on how this can be done.](https://github.com/basaks/uncover-ml/blob/master/uncoverml/geoio.py#L274)

**Comment by mcgarth** [comment]
Thanks for the suggestion Sudipta. Yesterday I implemented some compression using the GDAL library (Packbits filter). This can be seen in changes to shared.py in commit ccbe37c

This yielded a 2.2Gb reduction on a 5.5Gb full-res geotiff file. Interestingly it marginally *increased* the size of a multi-looked geotiff file (of 57 Mb size) comapred with no compression. Not ideal, but given the file size is very small its not too much of a concern right now.

A bigger change to the code that we might consider is actually skipping the first step of writing the full-res geotiff altogether. I investigated doing this but realised it is a bigger more complicated change than I have time to implement right now.
